### ISSUE TITLE
ui5-wizard: Screen Reader Support Enhancements

### ISSUE THREAD

**Issue by elenastoyanovaa** [SAP/ui5-webcomponents#2729]
Is your feature request related to a problem? Please describe.
The wizard screen reader support should be re-implemented, following the last recommendations and discussions with accessibility experts. The new proposal is described in the wiki page of the ui5-wizard component.

Additional context
After the implementation, update the wiki page with information regarding the speech output, link to the implementation PR and inform the accessibility expert.
